tests/internals/prometheus_metrics/prometheus_metrics_test.go

```diff
@@ -182,3 +250,36 @@ func testHPAScalerMetricValue(t *testing.T) {
 		t.Errorf("metric not available")
 	}
 }
+
+func testTriggerTotalMetric(t *testing.T, data templateData) {
+	t.Log("--- testing trigger total metric ---")
+	testTriggerTotalMetricValue(t, map[string]int{"kubernetes-workload": 1})
+
+	KubectlApplyWithTemplate(t, data, "cronScaledJobTemplate", cronScaledJobTemplate)
+	testTriggerTotalMetricValue(t, map[string]int{"kubernetes-workload": 1, "cron": 2})
+
+	KubectlDeleteWithTemplate(t, data, "cronScaledJobTemplate", cronScaledJobTemplate)
+	testTriggerTotalMetricValue(t, map[string]int{"kubernetes-workload": 1})
```

Comment:
Let's wait till the current e2e execution finishes, but I'd say that this test is flaky because there are other e2e tests running at the same time, so this values won't be exact, depending on which tests are running at the same time, these values will change.

Reply:
Hmm, you're right. When running standalone the test should work as expected, but it can fail when running the suite. I  don't know how to account for this however.

Reply:
I'd check if we have **at least** the expected values, if we have more, could be due to other e2e test running, but if we have less is because it's failing

Reply:
I would argue that the test is not really helpful in that case. There could be an error in the metric collection logic causing it to be incremented more than it should, and the test would fail to catch this.

Reply:
you are right...
Other option is getting current SO/SJ inside the test and evaluate them as part of the test, just comparing the metric value with the evaluated value you could know if it's working or not because you are getting the current objects during test execution

Reply:
The other way is to have a list of tests specified in the `run-all.sh` file, that are run in isolation.

Reply:
I know that we will duplicate logic, but if we change it by error in core code, the test will fail because the parsing logic is also in e2e test code

Reply:
> you are right... Other option is getting current SO/SJ inside the test and evaluate them as part of the test, just comparing the metric value with the evaluated value you could know if it's working or not because you are getting the current objects during test execution. WDYT?

I am not sure I understand what you mean by this.

Reply:
Do you mean fetching the total number of SO / SJ, counting the triggers manually and comparing them to the metric?

Reply:
yes, that's it

Reply:
The test can still get messed up because some scaled object might get created / deleted during the time we fetch the list and compare the metrics. But there's lesser room for error.

Reply:
Done.